dashboard/routes/node.lua

```diff
@@ -0,0 +1,324 @@
+        local is_admin = false
+
+        if req and req.session and req.session.get("user") then
+            is_admin = req.session.get('user').is_admin
+        end
+
+        if not is_admin then
+            return res:json({
+                success = false,
+                msg = "您的身份不是管理员，不允许创建节点."
+            })
+        end
+
+        local name = req.body.name
+        local ip = req.body.ip
+        local port = tonumber(req.body.port)
+        local api_username = req.body.api_username
+        local api_password = req.body.api_password
+
+        -- name
+        local pattern = "^[a-zA-Z][0-9a-zA-Z_]+$"
+        local match, err = smatch(name, pattern)
+
+        if not name or not ip or name == "" or ip == "" then
+            return res:json({
+                success = false,
+                msg = "节点名和IP不得为空."
+            })
+        end
+
+        local name_len = slen(name)
+        local ip_len = slen(ip)
+
+        if name_len < 1 or name_len > 20 then
+            return res:json({
+                success = false,
+                msg = "节点名长度应为1~20位."
+            })
+        end
+
+        if not match then
+            return res:json({
+                success = false,
+                msg = "节点名只能输入字母、下划线、数字，必须以字母开头."
+            })
+        end
+
+        -- ip
+        if ip_len < 7 and ip_len > 15 then
+            return res:json({
+                success = false,
+                msg = "IP 长度应为 7-15 位."
+            })
+        end
+
+        -- port
+        if port < 1 or port > 65535 then
+            return json:json({
+                success =false,
+                msg = "端口号为 1~65535 间的数字"
+            })
+        end
+
+
+        -- check ip
+        local result, err = node_model:query_by_ip(ip)
+        local isExist = false
+        if result and not err then
+            isExist = true
+        end
+
+        if isExist == true then
+            return res:json({
+                success = false,
+                msg = "该节点 IP 已添加到节点集群中"
+            })
+        else
+            -- save node info to db
+            local result, err = node_model:new(name, ip, port, api_username, api_password)
+            if result and not err then
+                return res:json({
+                    success = true,
+                    msg = "新建节点成功.",
+                    data = {
+                        nodes = node_model:query_all()
+                    }
+                })
+            else
+                return res:json({
+                    success = false,
+                    msg = "新建节点失败."
+                })
+            end
+        end
+    end)
+
+    node_router:post("/node/modify", function(req, res, next)
+        local is_admin = false
+
+        if req and req.session and req.session.get("user") then
+            is_admin = req.session.get('user').is_admin
+        end
+
+        if not is_admin then
+            return res:json({
```

Comment:
Use `http://lor.sumory.com/guide/multi_business_middleware.html` to reduce dumplicated authority judgement.